tempodb/backend/s3/s3.go

```diff
@@ -372,8 +376,13 @@ func createCore(cfg *Config, hedge bool) (*minio.Core, error) {
 		return nil, errors.Wrap(err, "create minio.DefaultTransport")
 	}
 
-	if cfg.InsecureSkipVerify {
-		customTransport.TLSClientConfig.InsecureSkipVerify = true
+	tlsConfig, err := cfg.GetTLSConfig()
```

Comment:
how does this work? i'm reading the code here:

https://github.com/grafana/dskit/blob/main/crypto/tls/tls.go#L87

and it looks like you have to set a reader somewhere?

Also, it looks like `GetTLSConfig()` always returns a value and so it will always be set on line 385 below. Is this ok? Should there be a way to not set this and just use defaults?

Reply:
Good find.  That commit isn't in the branch currently, but it looks like was introduced here: https://github.com/grafana/dskit/pull/274

I assume the reader interface was just to provide a little more flexibility on where to read the config data.

The nil check is just a habit when I see a pointer that could be nil, just in case the implementation changes we're still being defensive.  I can remove.

I'll double check, but it looked to me like if we don't set the config items, then we get a mostly default tls config, which looks okay to me.  Also note, that what we got back from the minio tls code is non-default, and so we set the min version in our config `RegisterFlagsAndApplyDefaults` to account for this.

Reply:
ok, rereading this makes way more sense. I missed that `GetTLSConfig()` is being called on the composed struct above. What is returned here will be the product of the settings. Wuf.